cmake/custom_cxxstandard.cmake

```diff
@@ -6,8 +6,8 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
     endif()
 
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Intel)
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16)
-        message(FATAL_ERROR "ICPC version must be at least 2016!")
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "16.0.2")
+        message(FATAL_ERROR "ICPC version must be at least 2016.0.2!")
```

Comment:
Supposedly [this works](https://github.com/psi4/psi4/blob/master/cmake/autocmake_omp.cmake#L55). So I'm hoping that the check here will allow 2016.0.2 and above and prevent the buggy 2016.0.1. I'll add a note asking for testers.